spec/model_spec.rb

```diff
@@ -160,6 +160,25 @@
     u.activated_state_at.should_not be_nil
   end
 
+  it "should skip tracking timestamps if opted out of with thread safety" do
+    threads = []
+    skip = User.new(email: "skip@example.com")
+    nope = User.new(email: "nope@example.com")
+
+    threads << Thread.new do
+      nope.semiactivate!
```

Comment:
I think you'd want to ensure the 2nd thread was within the block for this test to be valid. Something like:


```ruby
threads << Thread.new do 
  sleep 0.5
  nope.semiactivate!
end

threads << Thread.new do
  skip.without_state_transition_tracking do 
    sleep 1
    skip.semiactivate!
  end
end

threads.each(&:join)
```

Reply:
bleh thanks, good point. updating